lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -1320,20 +1320,7 @@ def get_networks_names_ids(self):
         return [{'name': n['Name'], 'id': n['Id']} for n in self.client.networks()]
 
     def get_service(self, name):
-        # The Docker API allows filtering services by name but the filter looks
-        # for a substring match, not an exact match. (Filtering for "foo" would
-        # return information for services "foobar" and "foobuzz" even if the
-        # service "foo" doesn't exist.) Avoid incorrectly determining that a
-        # service is present by filtering the list of services returned from the
-        # Docker API so that the name must be an exact match.
-        raw_data = [
-            service for service in self.client.services(filters={'name': name})
-            if service['Spec']['Name'] == name
-        ]
-        if len(raw_data) == 0:
-            return None
-
-        raw_data = raw_data[0]
+        raw_data = self.client.inspect_service(name)
```

Comment:
`inspect_service()` returns a `NotFound` exception if it cannot find the service ([CI error](https://app.shippable.com/github/ansible/ansible/runs/108392/51/tests)). Check out module_utils/docker/common.py, it catches that in a few cases I think.